src/services/storage/StorageItem.ts

```diff
@@ -0,0 +1,5 @@
+export abstract class StorageItem {
+  public abstract readonly id: string
+
+  public readonly storageTimestamp = new Date()
```

Comment:
This is important to allow the storage class to determine if a value is newer. Basically, since all flow runs are initialized with this timestamp we can safely check the timestamp when adding a flow run to the storage that already exists to see if its "newer". If its not, we don't overwrite the already newer flow run in the store.